### PR TITLE
add loaders while plugins, pipeplines and catalog fetches and making plugin selcetable without deselecting the previous selected one

### PR DESCRIPTION
**Pull request by mightyFZeus**
add loaders while plugins, pipeplines and catalog fetches

### CODE REVIEW

**Comment by mightyFZeus** [comment]
closes #654 ....A loader while pipeline, plugins and compute catalog fetches

Screenshot (103)




@PintoGideon

**Comment by mightyFZeus** [comment]
closes #689 .......plugins options are now selectable without having to deselect the pre-selected one
Screenshot (104)

File: package.json [hunk 1]
```diff
@@ -106,7 +106,6 @@
     "@types/uuid": "^8.3.0",
     "@typescript-eslint/eslint-plugin": "^5.37.0",
     "@typescript-eslint/parser": "^5.37.0",
-    "cypress": "^10.7.0",
```
Comment (jennydaman) [inline package.json:109]: why did you remove cypress?

File: src/components/catalog/PipelineCatalog.tsx [hunk 2]
```diff
@@ -15,6 +17,7 @@ const PipelineCatalog = () => {
 
   const { page, perPage, search } = pageState
   const [selectedPipeline, setSelectedPipeline] = React.useState<any>()
+  const [fetchingData, setFetchinData] = React.useState(false)
```
Comment (jennydaman) [inline src/components/catalog/PipelineCatalog.tsx:20]: The variable name `setFetchinData` is a little off-putting.

File: src/components/catalog/PipelineCatalog.tsx [hunk 3]
```diff
@@ -67,6 +72,9 @@ const PipelineCatalog = () => {
           }
         })
       }
+      setTimeout(() => {
+        setFetchinData(false)
+      }, 2000)
```
Comment (jennydaman) [inline src/components/catalog/PipelineCatalog.tsx:77]: How come the loading screen is hard-coded to be turned off after 2 seconds?

**Review by jennydaman — CHANGES_REQUESTED** [review]
According to the screenshots, these look like good changes. I had some questions above.

When working on codebases collaboratively, it's best to have one branch per feature. I see that you have added commits addressing multiple issues all to one branch called `olabisi`. It would be easier for us if you implemented independent features in independent branches, and created a PR for each feature branch.

File: package.json (hunk 1 above)
Comment (mightyFZeus) [reply package.json:109]: I installed cypress when I was working on the code coverage issue, then I had some issues running the Chris app cause I wasn't able to set it up, so I had to uninstall it so I could work on other issues. Initially, cypress wasn't installed, I had to install it so I could run the cypress script command in package.json

File: src/components/catalog/PipelineCatalog.tsx (hunk 3 above)
Comment (mightyFZeus) [reply src/components/catalog/PipelineCatalog.tsx:77]: I guess I was just trying to give the user a bit of a wait cause sometimes the data fetches rapidly and it looks like a bit of a snap. I would make the necessary Changes

**Comment by mightyFZeus** [comment]
Okay, I will fix up next time. Thank you

**Comment by mightyFZeus** [comment]
@jennydaman hi, so i  deleted the project on my laptop and cloned a new one so i can have a clean start, #723  closes #654
